langkit/templates/python_api/iterator_types_py.mako

```diff
@@ -0,0 +1,87 @@
+## vim: filetype=makopython
+
+<%def name="base_decl()">
+
+class _BaseIterator(object):
+    """
+    Base class for Ada iterator bindings.
+    """
+
+    c_element_type = None
+    """
+    Ctype class for iterator elements.
+    """
+
+    __slots__ = ('c_value',)
+
+    def __init__(self, c_value):
+        self.c_value = c_value
+
+    def __repr__(self):
+        return '<{}>'.format(type(self).__name__)
+
+    def clear(self):
+        self.c_value = None
```

Comment:
This method is for internal use: please add a `_` prefix (same for wrap and wrap_iter IIUC)

Reply:
Likewise for `c_value`

Reply:
and all the C types and function wrappers in subclasses :-)